### PR TITLE
makes barricades block weeds from spreading

### PR DESCRIPTION
**Pull request by MacBlaze1**
## About The Pull Request

this pr makes it so barricades block weeds from spreading onto them and past them. 

## Why It's Good For The Game

this will stop the cancerous action that xenos take of placing weeds then putting up walls super fast on barricades, then aciding the barricades.

## Changelog
:cl:
tweak: Barricades now block weeds from spreading onto and beyond the barricade turf
/:cl:

### CODE REVIEW

**Comment by Pariah919** [comment]
god be praised, marines can actually have map control

**Comment by Chaznoodles** [comment]
Ahem, based?

File: code/__HELPERS/patterns.dm [hunk 1]
```diff
@@ -71,6 +71,9 @@
 					continue
 				if (AdjT in results) // Ignore existing turfs
 					continue
+				if (barricade_block)
+					if (locate(/obj/structure/barricade) in AdjT)
+						continue
```
Comment (TiviPlus) [inline code/__HELPERS/patterns.dm:76]: put this in DirBlocked, dont use locate, check for density, remove barricade_block

**Review by TiviPlus — CHANGES_REQUESTED** [review]

**Comment by Denthamos** [comment]
Like the other resin/xeno building PR's up, #5045 and #5034 I feel like all three of these can all be traced back to the hivemind building resin into/past fobs and resin walls without marines being able to effectively stop it.  It seems like something needs to change there first due to how safe it is to just spam weeds and resin from that role without any repercussions.

File: code/__HELPERS/patterns.dm (hunk 1 above)
Comment (MacBlaze1) [reply code/__HELPERS/patterns.dm:76]: this would mess with flame grenades and other stuff If I didnt use a special variable

Comment (MacBlaze1) [reply code/__HELPERS/patterns.dm]: did your thing

File: code/game/objects/effects/weeds.dm [hunk 2]
```diff
@@ -165,7 +165,7 @@
 	update_icon()
 
 	// Generate our full graph before adding to SSweeds
-	node_turfs = filled_turfs(src, node_range, "square")
+	node_turfs = filled_turfs(src, node_range, "square", FALSE)
```
Comment (TiviPlus) [inline code/game/objects/effects/weeds.dm:168]:
```suggestion
	node_turfs = filled_turfs(src, node_range, "square")
```

File: code/__HELPERS/unsorted.dm [hunk 3]
```diff
@@ -206,6 +206,11 @@ GLOBAL_REAL_VAR(list/stack_trace_storage)
 	for(var/obj/structure/mineral_door/D in loc)
 		if(D.density)
 			return TRUE
+	for(var/obj/structure/barricade/D in loc)
+		if(!D.density)
+			continue
+		if(D.dir == direction)
+			return TRUE
```
Comment (TiviPlus) [inline code/__HELPERS/unsorted.dm:213]: change the copypaste D to a B

**Review by TiviPlus — CHANGES_REQUESTED** [review]

**Review by TiviPlus — APPROVED** [review]

**Pull request merged**